numerics/polynomial.hpp

```diff
@@ -76,9 +76,9 @@ class Polynomial {
   // making the destructor protected and nonvirtual.
   virtual ~Polynomial() = default;
 
-  virtual Value operator()(Argument const& argument) const = 0;
-  virtual Derivative<Value, Argument> EvaluateDerivative(
-      Argument const& argument) const = 0;
+  Value operator()(Argument const& argument) const;
+  Derivative<Value, Argument> EvaluateDerivative(
+      Argument const& argument) const;
```

Comment:
I am quite unhappy about this change.  `Polynomial` was supposed to be a pure interface used for heterogeneous storage in containers.  Having the FMA dispatching happen here looks like a code smell to me: conceptually we should be able to express polynomials in any base, and it's not clear that the FMA optimization is beneficial in all cases.

I would leave this class alone and do the FMA dispatching in `PolynomialInMonomialBasis` (in which case the with/without FMA functions don't need to be virtual, of course).